Changelog.md

```diff
@@ -7,12 +7,14 @@ Language Features:
 
 Compiler Features:
  * LSP: Add rudimentary support for semantic highlighting.
+ * Yul IR Code Generation: Improved copy routines for arrays with packed storage layout.
  * Yul Optimizer: Improve inlining heuristics for via IR code generation and pure Yul compilation.
 
 
 Bugfixes:
  * ABI Encoder: When encoding an empty string coming from storage do not add a superfluous empty slot for data.
  * Common Subexpression Eliminator: Process assembly items in chunks with maximum size of 2000. It helps to avoid extremely time-consuming searches during code optimization.
+ * Yul IR Code Generation: More robust cleanup in corner cases during memory to storage copies.
```

Comment:
The code seems correctly refactored but I did not notice any changes in functionality. Can you expand which corner cases were broken?

Reply:
Looking at the PR description again, I guess it's about the issues mentioned there. Maybe we should at least mention tuple cleanup here?

Reply:
It doesn't really fix anything about that, though. It just so happens that some corner cases are less broken if one uses more optimized copy routines, but other cases will still have the issue.
What the change does is merely to align which cases by chance happen to be fine between legacy and IR codegen (since it makes sense to have the optimization in IR codegen anyways).